src/dotnet-mono/Program.fs

```diff
@@ -156,12 +155,28 @@ module Main =
             |> logger.logSync
             Shell.killAllCreatedProcesses()
         )
-
+        
+    let splitArgs args =
+        let argus =
+            args
+            |> Array.takeWhile(fun x -> x <> "--")
+        let additionalArgs =
+            args
+            |> Array.skipWhile(fun x -> x <> "--")
+            |> Array.filter(fun x -> x <> "--")
```

Comment:
This would filter out all occurrences of `--`, which might not be desirable. It's conceivable that the program you're running with dotnet-mono would also want to use `--` in its parameter list, if it's trying to conform to the standard Unix arg-passing convention. E.g., if I've written a grep implementation in F# that I'm running with dotnet-mono, and I want to grep my source for the string `--help`, I might need to write a command line like `dotnet-mono MyGrep.exe -- --real-parameter -- --help`. With the current implementation using `Array.filter`, the second `--` would be stripped out and MyGrep would print its help text instead of grepping for the string `--help`.

I'd rather see this `Array.filter` line be:


```fsharp
|> (fun a -> if Array.tryHead a = Some "--" then Array.tail a else a)
```

That way if the parameters I need to pass to my program themselves include a `--`, it will still be possible to pass them.

Another way to write that logic, of course, would be:


```fsharp
|> (fun a -> if Array.length a > 0 && a.[0] = "--" then a |> Array.skip 1 else a)
```

Reply:
Ah yes good catch.

Reply:
Also I'm sure theres a more efficient way to do this but I don't think i'm concerned about looping twice with a few args.